Chapter-4-applying-tactical-domain-driven-design/Fitnet.Contracts/Src/Fitnet.Contracts.Core/Contract.cs

```diff
@@ -43,12 +43,20 @@ public static Contract Prepare(Guid customerId, int customerAge, int customerHei
             StandardDuration);
     }
 
-    public void Sign(DateTimeOffset signedAt, DateTimeOffset today)
+    internal BindingContract Sign(DateTimeOffset signedAt, DateTimeOffset dateNow)
     {
+        if (IsSigned)
+        {
+            throw new ContractAlreadySignedException();
+        }
+
         BusinessRuleValidator.Validate(
             new ContractCanOnlyBeSignedWithin30DaysFromPreparation(PreparedAt, signedAt));
 
         SignedAt = signedAt;
-        ExpiringAt = today.Add(Duration);
+        var expiringAt = dateNow.Add(Duration);
+        var bindingContract = BindingContract.Acceptance(Id, CustomerId, Duration, expiringAt);
```

Comment:
Acceptance is a very bad wording here, it should be verb - like accept. But it does not fit here.

Why not BindingContract.Start?

Reply:
Yeah, I've forgotten change this method name to something better, that was my initial thought. "Start" is good enough. Thanks for you feedback!